torch/cuda/__init__.py

```diff
@@ -498,35 +499,116 @@ def _strtoul(s: str) -> int:
         if not s:
             return -1
         for idx, c in enumerate(s):
-            if not c.isdigit():
+            if not (c.isdigit() or (idx == 0 and c in '+-')):
                 break
             if idx + 1 == len(s):
                 idx += 1
         return int(s[:idx]) if idx > 0 else -1
 
+    def parse_list_with_prefix(prefix):
+        rcs: List[str] = []
+        for elem in var.split(","):
+            # Repeated id results in empty set
+            if elem in rcs:
+                return cast(List[str], [])
+            # Anything other but prefix is ignored
+            if not elem.startswith(prefix):
+                break
+            rcs.append(elem)
```

Comment:
Need to check the existence of the corresponding device.

```bash
CUDA_VISIBLE_DEVICES="GPU-<exist-GPU-UUID>,GPU-<non-exist-GPU-UUID>,GPU-<exist-GPU-UUID>"
```

should be parsed as:

```bash
CUDA_VISIBLE_DEVICES="GPU-<exist-GPU-UUID>"
```

stop at the first error.

Reply:
Done in `_transform_uuid_to_ordinals`